tango/device_proxy.py

```diff
@@ -334,29 +339,36 @@ def __DeviceProxy__setattr(self, name, value):
 
     try:
         self.__refresh_cmd_cache()
-    except:
-        pass
+    except Exception as e:
+        if cause is None:
+            cause = e
 
     if name_l in self.__get_cmd_cache():
         raise TypeError('Cannot set the value of a command')
 
     try:
         self.__refresh_attr_cache()
-    except:
-        pass
+    except Exception as e:
+        if cause is None:
+            cause = e
 
     if name_l in self.__get_attr_cache():
         return __set_attribute_value(self, name, value)
 
     try:
         self.__refresh_pipe_cache()
-    except:
-        pass
+    except Exception as e:
+        if cause is None:
+            cause = e
 
     if name_l in self.__get_pipe_cache():
         return self.write_pipe(name, value)
 
-    return super(DeviceProxy, self).__setattr__(name, value)
+    try:
+        return super(DeviceProxy, self).__setattr__(name, value)
+    except Exception as e:
+        raise e from cause
```

Comment:
This case is interesting.  If we get here then `name` isn't a known Tango device command/attribute/pipe.  We end up writing to either an existing Python class instance attribute, or adding a new one, if it doesn't exist yet.

I'm not sure that `__setattr__` can even raise an exception?  If it did, the `cause` may or may not be relevant.

A couple of cases could occur for the example of an unreachable device:
1. We haven't yet cached the lists of commands, attributes and pipes.  
2. We have cached the lists of commands, attributes and pipes prior to this call. 

Then we get these four scenarios, depending on if the Tango command/attribute/pipe either:

- Exists on the device:
  - For case 1:  we'll write to a local class instance attribute and the user won't even get an error.  This behaviour is unexpected, but not a new problem.
  - For case 2: we would have failed because it is a command, or we tried to call `__set_attribute_value` or `write_pipe` earlier in this method, so the user will get whatever error those methods throw.  That's fine.
- Doesn't exist on the device:
  - For case 1:  we'll write to a local class instance attribute and the user won't get an error.  This could be the desired behaviour, if the user was storing some other info on an instance of a `DeviceProxy`.  Or, it could be undesired, if the user thought they were setting a Tango attribute/pipe but had the wrong name.  Not much we can do about that. 
  - For case 2: same as case 1.  There won't be an error, and we're not sure if the user expected one of not.

In conclusion, the change in the PR is fine.  It has just made us aware of some nuances in how setting attributes on a `DevicePrxoy` behaves.